src/Java/gtPlusPlus/xmod/gregtech/common/tileentities/machines/multi/production/GregtechMetaTileEntityTreeFarm.java

```diff
@@ -99,21 +100,28 @@ public String getMachineType() {
 		return "Tree Farm";
 	}
 
-	public String[] getTooltip() {
-
+	@Override
+	protected GT_Multiblock_Tooltip_Builder createTooltip() {
 		if (mCasingName.toLowerCase().contains(".name")) {
 			mCasingName = ItemUtils.getLocalizedNameOfBlock(ModBlocks.blockCasings2Misc, 15);
 		}
-
-		return new String[]{
-                "[WIP] Disabled",
-				"Converts EU to Oak Logs",
-				"Speed: Very Fast | Eu Usage: 100% | Parallel: 1",				
-				"Requires a Saw, Buzz Saw or Chainsaw in GUI slot",
-				"Constructed exactly the same as a normal Vacuum Freezer",
-				"Use "+mCasingName+"s (10 at least!)",
-				"TAG_HIDE_HATCHES"
-		};
+		GT_Multiblock_Tooltip_Builder tt = new GT_Multiblock_Tooltip_Builder();
+		tt.addMachineType(getMachineType())
+				.addInfo("Converts EU to Oak Logs")
+				.addInfo("Eu Usage: 100% | Parallel: 1")
+				.addInfo("Requires a Saw or Chainsaw in GUI slot")
+				.addPollutionAmount(getPollutionPerTick(null) * 20)
+				.addSeparator()
+				.beginStructureBlock(3, 3, 3, true)
+				.addController("Front center")
+				.addCasingInfo("Sterile Farm Casing", 10)
```

Comment:
Isn't localized, hence why mCasingName is set at the start of the tooltip.

Reply:
I tried using that but it was not working properly.
So I copied the implementation that was done on the rest of the multiblocks, unlocalized.